x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import {
+  RequestHandlerContext,
+  KibanaRequest,
+  KibanaResponseFactory,
+  Logger,
+} from 'src/core/server';
+import { ConfigType } from '../index';
+
+interface IEnterpriseSearchRequestParams<ResponseBody> {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData?: (body?: ResponseBody) => boolean;
+}
+
+/**
+ * This helper function creates a single standard DRY way of handling
+ * Enterprise Search API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
```

Comment:
Another quick App Search copy swap, + body was repeated twice so I pruned that out
```suggestion
 * This handler assumes that it will essentially just proxy the
 * Enterprise Search API request, so the request body and request
 * parameters are simply passed through.
```